app/workers/kms_key_rotation/batch_initiator_job.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module KmsKeyRotation
+  class BatchInitiatorJob
+    include Sidekiq::Worker
+
+    # sidekiq_options retry: 5, queue: :low
+    sidekiq_options retry: 5, queue: :default
+
+    RECORDS_PER_BATCH = 10
+    RECORDS_PER_JOB = 1
+    # RECORDS_PER_BATCH = 10_000
+    # RECORDS_PER_JOB = 1_000
+    # JOBS_PER_BATCH = 10
+
+    class BatchComplete
+      def on_complete(_status, _options)
+        BatchInitiatorJob.perform_async
+      end
+    end
+
+    def perform
+      batch = Sidekiq::Batch.new
+      batch.description = "KMS Key Rotation #{batch.bid}"
+      batch.on(:complete, BatchComplete)
+
+      return nil if records.empty?
+
+      begin
+        batch.jobs do
+          Sidekiq::Client.push_bulk('class' => KmsKeyRotation::RotateKeysJob, 'args' => batched_gids)
+        end
+      rescue => e
```

Comment:
I'm not sure we should be rescuing all errors here. 😬

Reply:
I was on the fence. I was thinking that If it threw an error for a one-off issue, then it would be nice to rescue it so that the job would keep going... However I guess if it never pushed the batch, it would never complete, and wouldn't continue anyway. So yeah, I agree.